### ISSUE TITLE
Upstart service provider should provide a way to restart with a stop + start

### ISSUE THREAD

**Issue by mconigliaro** [chef/chef#4877]
## Description

According to [the Upstart docs](http://upstart.ubuntu.com/cookbook/#restart):

> restart is not the same as running stop followed by start since the restart command will retain the original job configuration whereas stopping the job and restarting it will load the latest job configuration from disk.

This means notifying a service to restart after updating the Upstart job configuration may not work the way you expect. You might be able to work around this problem with the `restart_command` property, but it would be a lot simpler if something like a `:stop_start` action was available.

This was previously reported in [CHEF-4829](https://tickets.opscode.com/browse/CHEF-4829).
## Chef Version

```
# chef-client --version
Chef: 12.9.38
```
## Platform Version

```
# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.3 LTS
Release:    14.04
Codename:   trusty
```
## Replication Case

Updating the template will cause the service to be restarted, but not with the updated job configuration:

```
template '/etc/init/foo.conf' do
  notifies :restart, 'service[foo]'
end

service 'foo'
```

**Comment by coderanger** [comment]
Obligatory plug for poise-service: https://github.com/poise/poise-service/blob/master/lib/poise_service/service_providers/upstart.rb#L38-L46

That said, you can do this:

```ruby
template '/etc/init/foo.conf' do
  notifies :stop, 'service[foo]'
  notifies :start, 'service[foo]'
end
```

**Comment by mconigliaro** [comment]
@coderanger 👍  for the slightly better workaround. After some Googling, it looks like maybe `supports restart: false` might also force a stop + start?

**Comment by thommay** [comment]
Fixed in #5063
